doc/contributing.rst

```diff
@@ -1,221 +1,760 @@
+  Windows 10 also offers the `Windows subsystem for Linux`_ that offers similar
+  functionality to git Bash, but has not been widely tested by MNE-Python
+  developers yet.
 
-* Use `numpy style`_ for docstrings. Follow existing examples for simplest
-  guidance.
+.. note::
 
-* New functionality must be covered by tests. For example, a
-  :class:`mne.Evoked` method in ``mne/evoked.py`` should have a corresponding
-  test in ``mne/tests/test_evoked.py``.
+    `GitHub desktop`_ is a GUI alternative to command line git that some users
+    appreciate; it is available for Windows and MacOS.
 
-* Changes must be accompanied by updated documentation, including
-  :doc:`doc/whats_new.rst <whats_new>` and
-  :doc:`doc/python_reference.rst <python_reference>`.
+Once git is installed, the only absolutely necessary configuration step is
+identifying yourself and your contact info:
 
-* After making changes, **ensure all tests pass**. This can be done
-  by running:
+.. code-block:: console
 
-  .. code-block:: console
+   $ git config --global user.name "Your Name"
+   $ git config --global user.email you@yourdomain.example.com
 
-     $ make test
+Make sure that the same email address is associated with your GitHub account
+and with your local git configuration. It is possible to associate multiple
+emails with a GitHub account, so if you initially set them up with different
+emails, just add the local email to the GitHub account.
 
-  To run individual tests, you can also run e.g.:
+Sooner or later, git is going to ask you what text editor you want it to use
+when writing commit messages, so you might as well configure that now too:
 
-  .. code-block:: console
+.. code-block:: console
 
-     $ pytest mne/tests/test_evoked.py:test_io_evoked -x --verbose
+   $ git config --global core.editor emacs    # or vim, or nano, or subl, or...
 
-  Or alternatively:
-  
-  .. code-block:: console
+There are many other ways to customize git's behavior; see `configuring git`_
+for more information. Once you have git installed and configured, and before
+creating your local copy of the codebase, go to the `MNE-Python GitHub`_ page
+and create a `fork`_ into your GitHub user account.
 
-     $ pytest mne/tests/test_evoked.py -k test_io_evoked -x --verbose
 
-  Make sure you have the testing dataset, which you can get by doing::
+Setting up the Python environment
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-     >>> mne.datasets.testing.data_path(verbose=True)  # doctest: +SKIP
+.. note::
 
-MNE-specific coding guidelines
-------------------------------
+    We strongly recommend the `Anaconda`_ or `Miniconda`_ environment managers
+    for Python. Other setups are possible but are not officially supported by
+    the MNE-Python development team; see discussion :ref:`here
+    <other-py-distros>`.
 
-These are guidelines that are generally followed:
+The first step is to `clone`_ the MNE-Python repository from your fork, and
```

Comment:
Once you have a working Python environment, the first step ...

And some part of this could link to our install doc